manifests/params.pp

```diff
@@ -269,12 +269,16 @@
   $dhcp_load_balance = 3
 
   # DNS settings - requires optional DNS puppet module
+  if fact('networking.domain') {
+    $dns_zone = $facts['networking']['domain']
+  } else {
+    $dns_zone = ''
+  }
   $dns                    = false
   $dns_listen_on          = 'https'
   $dns_managed            = true
   $dns_provider           = 'nsupdate'
   $dns_interface          = pick(fact('networking.primary'), 'eth0')
-  $dns_zone               = $facts['networking']['domain']
   $dns_realm              = upcase($dns_zone)
```

Comment:
This is only used in `$dns_tsig_principal`. I don't know what a sane default for that is if there is no domain. Is there any?

Reply:
`$dns_tsig_principal` is only used to manage an external DNS server with Kerberos. I would say if there is no domain in that scenario, your problems are much bigger. Therefore I think it's fine to fallback on a dummy domain like `example.com` when the domain can't be found ? (since the empty string is not great for reasons you already described)